modified the README in roles/pool dir (#802)
stratum-mining/stratum: roles/pool/README.md

```

## Setup

### Configuration File

`pool-config-hosted-tp-example.toml` and `pool-config-local-tp-example.toml` are examples of configuration files.

The configuration file contains the following information:

> 1. The SRI Pool information which includes the SRI Pool authority public key
>    (`authority_public_key`), the SRI Pool authority secret key (`authority_secret_key`), along with its certificate validity (`cert_validity_sec`). In addition to this, it contains the address which it will use to listen to new connection from downstream roles (`listen_address`),it has the list of uncompressed pubkeys for coinbase payout (`coinbase_outputs`) and a string that serves as coinbase tx known as (`pool_signature`).

Reviewer: ```suggestion
1. The SRI Pool information which includes the SRI Pool authority public key
   (`authority_public_key`), the SRI Pool authority secret key (`authority_secret_key`).
 2. The address which it will use to listen to new connection from downstream roles (`listen_address`)
 3. The list of uncompressed pubkeys for coinbase payout (`coinbase_outputs`)
 4. A string that serves as signature on the coinbase tx (`pool_signature`).
```

```

## Setup

### Configuration File

> The `pool-config-hosted-tp-example.toml` is a configuration example directory by the party that is running the SV2 Pool (most
> typically the pool service provider) to address the most preferred customization.

Reviewer: ```suggestion
`pool-config-hosted-tp-example.toml` and `pool-config-local-tp-example.toml` are examples of configuration files.
```

```

## Setup

### Configuration File

`pool-config-hosted-tp-example.toml` and `pool-config-local-tp-example.toml` are examples of configuration files.

The configuration file contains the following information:

> 1. The SRI Pool information which includes the SRI Pool authority public key
>    (`authority_public_key`), the SRI Pool authority secret key (`authority_secret_key`).
>  2. The address which it will use to listen to new connection from downstream roles (`listen_address`)
>  3. The list of uncompressed pubkeys for coinbase payout (`coinbase_outputs`)
>  4. A string that serves as signature on the coinbase tx (`pool_signature`).

Reviewer: ```suggestion
1. The SRI Pool information which includes the SRI Pool authority public key
   (`authority_public_key`), the SRI Pool authority secret key (`authority_secret_key`).
2. The address which it will use to listen to new connection from downstream roles (`listen_address`)
3. The list of uncompressed pubkeys for coinbase payout (`coinbase_outputs`)
4. A string that serves as signature on the coinbase tx (`pool_signature`).
```

> 2. The SRI Pool Job Negatiator information which includes the Template Provider address (`tp_address`) and the address it uses to listen new request from the downstream JDs (`jd_address`).

Reviewer: ```suggestion
5. The Template Provider address (`tp_address`).
```

> 3. Optionally, you may want to verify that your TP connection is authentic. You may get `tp_authority_public_key` from the logs of your TP, for example:

Reviewer: ```suggestion
6. Optionally, you may want to verify that your TP connection is authentic. You may get `tp_authority_public_key` from the logs of your TP, for example:
```

```
# 2024-02-13T14:59:24Z Template Provider authority key: EguTM8URcZDQVeEBsM4B5vg9weqEUnufA8pm85fG4bZd
```

### Run

> There are two files found in cd roles/pool/config-examples
> 
> 1. pool-config-hosted-tp-example.toml runs on our community hosted server.
> 2. pool-config-example-tp-example.toml runs with your local config.

Reviewer: ```suggestion
There are two files found in `roles/pool/config-examples`

1. `pool-config-hosted-tp-example.toml` runs on our community hosted server.
2. `pool-config-example-tp-example.toml` runs with your local config.
```

Run the Pool:

```bash

>   cd roles/pool/config-examples
>   cargo run -- -c pool-config-hosted-tp-example.toml

Reviewer: ```suggestion
cd roles/pool/config-examples
cargo run -- -c pool-config-hosted-tp-example.toml
```
